test/mysql.test.js

```diff
@@ -21,6 +22,13 @@ ObjectID.prototype.toJSON = function() {
   return this.id1 + this.id2;
 };
 
+var deleteModels = function(models, done) {
+  async.each(
+      models,
+      function(Model, next) { Model.destroyAll(next); },
+        done);
```

Comment:
I think this line is indented too much, you should remove two spaces.

Also `deleteModel` should be defined as a regular function, the code should be placed towards the end of the `describe` block, after the tests. See http://loopback.io/doc/en/contrib/style-guide.html#layout-of-test-files

BTW since we can use ES6 and promises in `master`, the code can be simplified even more:

```js
// in your hook
beforeEach(function() {
  return deleteAllModelInstance([...]);
});

// function definition
function deleteAllModelInstances(models) {
  return Promise.all(models.map(m => m.destroyAll());
}
```

AFAICT, the list of `models` we are deleting is always the same, isn't it? In which case we can make `deleteAllModelInstance` parameter-less, which was what I had originally in mind.

```js
// at the top
beforeEach(deleteAllModelInstances);

// at the bottom
function deleteAllModelInstances() {
  const models = [...];
  return Promise.all(models.map(m => m.destroyAll());
}

// or perhaps
function deleteAllModelInstances() {
  return Promise.all([
    PostWithDefaultId.destroyAll(),
    // ...
    Student.destroyAll(),
  ]);
```

Thoughts?

Let's not waste too much time on this code cleanup - pick whatever makes most sense to you.